op-service/sources/prefetching_eth_client.go

```diff
@@ -30,34 +31,41 @@ type EthClientInterface interface {
 }
 
 type PrefetchingEthClient struct {
-	inner            EthClientInterface
-	PrefetchingRange uint64
+	inner              EthClientInterface
+	PrefetchingRange   uint64
+	PrefetchingTimeout time.Duration
+	runningCtx         context.Context
```

Comment:
the context should be created within `NewPrefetchingEthClient` and then saved to the two fields
```suggestion
	runningCtx         context.Context
	runningCancel   context.CancelFunc
```
and then call `p.runningCancel()` in `Close()`.

the context should not be passed as an argument to `NewPrefetchingEthClient` otherwise it would cancelled already when the function calling the ctor returns.

Reply:
You're right, I thought it'd be fine if the constructor context gets passed in/and then handled a specific way - but why even worry about that when we can just create one in the constructor?